src/test/test-local-addresses.c

```diff
@@ -40,5 +40,12 @@ int main(int argc, char *argv[]) {
         print_local_addresses(a, (unsigned) n);
         free(a);
 
+        n = local_outbounds(NULL, 0, AF_UNSPEC, &a);
+        assert_se(n >= 0);
```

Comment:
will this test fail if the tests are run in some unprivileged, network-less, constrained container?

Reply:
well, this PR doesn't add much new in that area. i.e. we first invoke local_gateways() (which the test calls anyway already), and then issue a connect() to each gateway, and ignore almost all errors on those socket operations. Thus, if the test previously was not an issue it shouldn't be an issue now either. The security-wise problematic stuff is more likely the rtnetlink stuff, but so far this proved not to be a problem. The UDP socket stuff we do should be unproblematic, in particular as we — as mentioned — ignore almost all errors for it, or handle them gracefully